task_generator/task_generator/manager/entity_manager/crowdsim_manager.py

```diff
@@ -465,7 +466,7 @@ def remove_obstacles(self, purge):
 
             for obstacle_id, obstacle in self._known_obstacles.items():
                 if purge >= obstacle.layer:
-                    if isinstance(self._simulator, GazeboSimulator):
```

Comment:
reconsider function structuring, might be better to write a new function for case:
if isinstance(self._simulator, GazeboSimulator) or isinstance(self._simulator, UnitySimulator):

Reply:
@tuananhroman I get that. But is this case it's a large function and there is only one block where Gazebo/Unity is distinguished from Flatland. I don't think it makes sense to split this function.
Since I didn't write this function and it says "CRITICAL REGION" I believe it's better to leave as is. @voshch

Reply:
It's written this way since deleting certain models (those containing pedsim plugins) in Gazebo crashes the plugin (race condition during destruction) and breaks the simulation. This shouldn't present a problem in unity, you can likely use the non-Gazebo case.

Reply:
'critical region' just means that it's guarded against very fast back-to-back resets that would interfere with each other.